src/wallet/export/caravan.rs

```diff
@@ -0,0 +1,461 @@
+            CaravanNetwork::Testnet => Network::Testnet,
+        }
+    }
+    /// Get the descriptor value
+    pub fn descriptor(&self) -> Result<Descriptor<DescriptorPublicKey>, Error> {
+        let required = self.quorum.required_signers;
+        let network: Network = self.network();
+
+        let result = match self.address_type {
+            CaravanAddressType::P2sh => {
+                let keys: Vec<DescriptorKey<Legacy>> = self.descriptor_keys()?;
+                descriptor! { sh ( sortedmulti_vec(required, keys) ) }
+            }
+            CaravanAddressType::P2shP2wsh => {
+                let keys: Vec<DescriptorKey<Segwitv0>> = self.descriptor_keys()?;
+                descriptor! { sh ( wsh ( sortedmulti_vec(required, keys) ) ) }
+            }
+            CaravanAddressType::P2wsh => {
+                let keys: Vec<DescriptorKey<Segwitv0>> = self.descriptor_keys()?;
+                descriptor! { wsh ( sortedmulti_vec(required, keys) ) }
+            }
+        }
+        .map_err(|e| Error::Descriptor(e));
+
+        match result {
+            Ok((d, _, n)) => {
+                if n.contains(&network) {
+                    Ok(d)
+                } else {
+                    Err(Error::InvalidNetwork {
+                        requested: network,
+                        found: *n.iter().last().expect("network"),
+                    })
+                }
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    fn descriptor_keys<Ctx: ScriptContext>(
+        &self,
+    ) -> Result<Vec<DescriptorKey<Ctx>>, DescriptorError> {
+        let result = self
+            .extended_public_keys
+            .iter()
+            .map(|k| {
+                let fingerprint = k.xfp;
+                let key_path = k.clone().bip32_path;
+                let mut key_source = None;
+                if let (Some(fp), Some(kp)) = (fingerprint, key_path) {
+                    key_source = Some((fp, kp))
+                };
+                let derivation_path =
+                    DerivationPath::master().child(ChildNumber::Normal { index: 0 });
+                k.xpub
+                    .into_descriptor_key(key_source, derivation_path)
+                    .map_err(|e| DescriptorError::Key(e))
+            })
+            .collect();
+        result
+    }
+
+    fn parse_sorted_multi<Pk: MiniscriptKey, Ctx: ScriptContext>(
+        sorted_multi: &SortedMultiVec<Pk, Ctx>,
+    ) -> (Quorum, Vec<Pk>) {
+        let quorum = Quorum {
+            required_signers: sorted_multi.k,
+            total_signers: sorted_multi.pks.len(),
+        };
+        let extended_public_keys = sorted_multi.pks.clone();
+        (quorum, extended_public_keys)
+    }
+
+    /// Export BDK wallet configuration as a Caravan configuration
+    pub fn export(
+        name: String,
+        client_type: String,
+        network: Network,
+        descriptor: &Descriptor<DescriptorPublicKey>,
```

Comment:
I think it would be better if you could take a reference to the wallet and get the descriptor yourself. It would also give you a chance to inspect the internal descriptor like we do for fully noded.

I'm guessing the caravan wallet only supports similar descriptors for the internal keychain

Reply:
I also have it on my issues list to determine if Caravan can support change descriptors, and if so how. I expect all I need to do is change the derivation from `m/0/*` to `m/1/*` but need to confirm.

Reply:
Caravan will create a change and a receive descriptor based on the settings imported into it. We call this data structure a "braid", since once it's not a string anymore it's not really a descriptor anymore. Basically nearly all wallets today contain two braids, one at index `0` and one at `1`. I'd also like to support [BIP88](https://github.com/bitcoin/bips/blob/master/bip-0088.mediawiki) style, but this should be done separate from this utility imo. i.e. in JS expand the single bip88 descriptor into two (or more?) descriptors and then feed that through bdk.